### ISSUE TITLE
env interface

### ISSUE THREAD

**Issue by mhhf** [nexusdev/dapple#237]
Using environments sometimes feels clumsy, maybe engineering an interface for an dapple module is suited. Here is a small draft as a starting point:

1. Environments are written in in `.dapplerc` and `dappfile` and follow the same [spec](https://github.com/nexusdev/dapple/blob/master/specs/definitions.schema.json#L98).
2. On execution, they are inherited: `.dapplerc <-- dappfile <-- cli flags`.
3. Environments itself can inherit from other environments: `morden <-- contractSystemOnMorden`
4. Environment manipulations are done in `dappfile` by default and can be exported to `.dapplerc`.
5. Environment redesign will include typed object renaming following the description in Issue #227 
6. Chain-Connectors will be redesigned according to this discussion: #160

### Capabilities:
* select environment (make default)
* list environments
* inspect environment
* fork environment
* new environment
* persistently save an environment

#### select:
`dapple env select <environment>`:
Makes the marked environment as default. Same behavior as adding `-e <environment>` flag to any command.  

#### list
`dapple env list`
Lists all available environments with minimal info.
```
morden -- ethereum: morden, 4 objects
homestead -- ethereum: homestead, 4 objects
testenv: -- inherits: morden, ethereum: morden, 7 objects
```

#### inspect
`dapple env inspect <environment>`
print data of the current environment, such as ethereum connection and status, ipfs connection, objects, registries, etc.

#### fork
`dapple fork <from> <to>`
create a new envoronment <to> by cloning <from> environment.

#### new
`dapple env new <environment>`
Creates a new environment by inheriting the current default.

#### save
`dapple env save`
Saves the current default environment to the users `.dapplerc`

Thoughts or thumbs?

**Comment by mhhf** [comment]
I think that environments are equivalent to chain states and should be tightly coupled with #239 - basically the notion of an environment is exactly the head state of DAG Node. I'll think how to merge this with #239

**Comment by mhhf** [comment]
Implemented in v0.8 through dapple-chain and dapple-script
